crates/libs/core/src/runtime/stateful.rs

```diff
@@ -95,8 +103,17 @@ impl From<&IFabricStatefulServicePartition> for StatefulServicePartition {
 /// TODO: replicator has no public documentation
 #[trait_variant::make(Replicator: Send)]
 pub trait LocalReplicator: Send + Sync + 'static {
-    async fn open(&self, cancellation_token: CancellationToken) -> crate::Result<crate::HSTRING>; // replicator address
+    /// Opens replicator, and returns the replicator address that is visible to primary
+    /// in ReplicaInformation.
+    /// Remarks:
+    /// Replicator does not have an assigned role yet and should setup listening endpoint.
+    async fn open(&self, cancellation_token: CancellationToken) -> crate::Result<crate::HSTRING>;
     async fn close(&self, cancellation_token: CancellationToken) -> crate::Result<()>;
+
+    /// Change the replicator role.
+    ///
+    /// Remarks:
+    /// Replicator change_role is called before Replica change_role.
```

Comment:
Is there any broader commentary to offer on the relationship between `Replica` and `Replicator`?

Reply:
Replica is for setting up listener for user connections and manage its life cycle. 
Replicator is for replication listener and LSN management in the background.